statsd.go

```diff
@@ -25,110 +26,159 @@ func init() {
 	}
 }
 
-func Timer() timer {
-	return timer{time.Now()}
+// Options holds key/value pairs to be used when calling the functions
+// with the "...WithOptions" suffix.
+type Options struct {
+	// Rate specifies the sampling rate to use.
+	Rate float64
+	// AlwaysSend specifies whether the packet should be sent
+	// to the server even if the specified sampling rate is too low.
+	AlwaysSend bool
+}
+
+// DefaultOptions returns the default options
+func DefaultOptions() *Options {
+	return &Options{
+		Rate:       1.0,
+		AlwaysSend: false,
+	}
 }
 
 // Timer
 type timer struct {
 	start time.Time
 }
 
+// Timer returns a new timer set to `time.Now()`
+func Timer() timer {
+	return timer{time.Now()}
+}
+
+// Reset sets the start time for the timer to `time.Now()`
 func (t *timer) Reset() {
 	t.start = time.Now()
 }
 
+// Send takes a list of remote timer names, and submits the time that
+// has ellapsed since the creation of the timer to each in turn.
+// It returns a time.Duration representing the amount of time that was sent.
 func (t *timer) Send(names ...interface{}) (took time.Duration) {
+	return t.SendWithOptions(nil, names...)
+}
+
+// SendWithOptions works like Send but sends the timing information
+// using the provided options.
```

Comment:
I've added a bunch of examples, and tested those with a local version of `godoc`. In order for the `timer.Send` functions to render, though, I had to export the struct, which I did as `Timing`, to avoid the conflict with the `Timer` function.